***HEART ATTACK ANALYSIS AND PREDICTION***

By use of heart.csv dataset, we will try to predict the heart attack based on the given features on the dataset.

In [4]:
import pandas as pd

data = pd.read_csv("./heart.csv")

print(data.head(10))

   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   52    1   0       125   212    0        1      168      0      1.0      2   
1   53    1   0       140   203    1        0      155      1      3.1      0   
2   70    1   0       145   174    0        1      125      1      2.6      0   
3   61    1   0       148   203    0        1      161      0      0.0      2   
4   62    0   0       138   294    1        1      106      0      1.9      1   
5   58    0   0       100   248    0        0      122      0      1.0      1   
6   58    1   0       114   318    0        2      140      0      4.4      0   
7   55    1   0       160   289    0        0      145      1      0.8      1   
8   46    1   0       120   249    0        0      144      0      0.8      2   
9   54    1   0       122   286    0        0      116      1      3.2      1   

   ca  thal  target  
0   2     3    0.23  
1   0     3    0.37  
2   0     3    0.24  
3   1     3    0.28 

**Age**: Numeric (e.g., 52)
**Sex**: Categorical (0: Female, 1: Male)
**Chest Pain Type**: Categorical (0: Typical Angina, 1: Atypical Angina, 2: Non-anginal Pain, 3: Asymptomatic)
**Resting Blood Pressure**: Numeric (e.g., 125)
**Serum Cholesterol**: Numeric in mg/dL (e.g., 212)
**Fasting Blood Sugar**: Categorical (0: <= 120 mg/dL, 1: > 120 mg/dL)
**Resting Electrocardiographic Results**: Categorical (0: Normal, 1: Abnormality, 2: Hypertrophy)
**Maximum Heart Rate Achieved**: Numeric (e.g., 168)
**Exercise-Induced Angina**: Categorical (0: No, 1: Yes)
**Oldpeak (ST Depression)**: Numeric (e.g., 1.0)
**Slope of Peak Exercise ST Segment**: Categorical (0: Upsloping, 1: Flat, 2: Downsloping)
**Number of Major Vessels Colored by Fluoroscopy**: Numeric (0 to 3)
**Thalassemia**: Categorical (0: Normal, 1: Fixed Defect, 2: Reversible Defect)